internal/mode/static/state/relationship/capturer.go

```diff
@@ -126,6 +103,10 @@ func (c *CapturerImpl) Capture(obj client.Object) {
 			gateways: gateways,
 		}
 		c.namespaces[client.ObjectKeyFromObject(o)] = nsCfg
+	case *v1beta1.GatewayClass:
+		if o.Spec.ParametersRef != nil && string(o.Spec.ControllerName) == c.gwCtlrName {
```

Comment:
Can we leverage the gateway class name that is passed in via cli args here? I don't think we care about all the GatewayClasses with our controller name, just the one that we are configured to watch.

Reply:
Ah yes, forgot we had that.